scripts/parse_full_results.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# coding: utf8
+'''Script to parse raw annotation results into a sparse label collection.
+
+Example
+-------
+$ ./scripts/parse_raw_results.py "path/to/dir/*.csv" sparse-labels.csv
+
+'''
+from __future__ import print_function
+
+import argparse
+import glob
+import os
+import pandas as pd
+import sys
+import tqdm
+import uuid
+import warnings
+
+YN_MAP = {'no': 0, 'yes': 1}
+OUTPUT_COLUMNS = ['sample_key', 'worker_id', 'worker_trust',
+                  'channel', 'instrument', 'response']
+CONTAIN_COL = 'does_this_recording_contain_{}'
+
+
+def parse_one(row):
+    '''
+    Parameters
+    ----------
+    row : pd.Series
+        Series record with at least the following fields:
+          [_channel, _trust, _worker_id, instrument, sample_key,
+           does_this_recording_contain_{instrument}]
+
+    Returns
+    -------
+    resp : dict
+        Object with the following fields:
+          [sample_key, worker_id, worker_trust, channel, instrument, response]
+    '''
+
+    response = YN_MAP.get(row[CONTAIN_COL.format(row.instrument)])
+    if response is None:
+        warnings.warn("Null response: {}".format(row.tolist()))
+
+    return dict(sample_key=row.sample_key, worker_id=row._worker_id,
+                worker_trust=row._trust, channel=row._channel,
+                instrument=row.instrument, response=response)
+
+
+def encrypt_field(values, hashlen=8, retries=5):
+    unique_values = set(values)
+    for n in range(retries):
+        hashmap = {val: str(uuid.uuid4()).replace('-', '')[:hashlen]
+                   for val in list(unique_values)}
+
+        if len(hashmap) == len(unique_values):
+            break
```

Comment:
Why would this condition ever happen?  `len(dict) == len(dict.keys()) == len(set(dict.keys()))`.  Did you mean to check the number of unique hash outputs here?

Reply:
yea, good catch. that was the thinking, as a check on accidentally creating key collisions. Definitely a bug tho, should be `len(set(hashmap.values()))`, will fetch